salt/utils/url.py

```diff
@@ -47,7 +47,10 @@ def create(path, saltenv=None):
 
     query = f"saltenv={saltenv}" if saltenv else ""
     url = salt.utils.data.decode(urlunparse(("file", "", path, "", query, "")))
-    return "salt://{}".format(url[len("file:///") :])
+    # urlunparse changed behavior in Python 3.13
+    if url.startswith("file:///"):
+        return "salt://{}".format(url[len("file:///") :])
+    return "salt://{}".format(url[len("file:") :])
```

Comment:
This create function should not be doing `path = path.replace("\\", /)` and probably not 
```python
if salt.utils.platform.is_windows():
path = salt.utils.path.sanitize_win_path(path)
```
Instead it should use `urllib.request.pathname2url` or some other python builtin functionality.

Especially since the `sanitize_win_path` function is clearly miswritten:

```python
def sanitize_win_path(winpath):
    """
    Remove illegal path characters for windows
    """
    intab = "<>:|?*"
    if isinstance(winpath, str):
        winpath = winpath.translate({ord(c): "_" for c in intab})
    elif isinstance(winpath, str):
```
Notice the duplicate `isinstance(winpath, str)` in the elif that will never occur.

I would prefer instead of assuming that the string starts with `file:` or `file:///` we do a regex search replace like `re.sub(r"^file:(?:///)?", "", url)` Also use `f` string instead of `format`.

```python
return f'salt://{re.sub(r"^file:(?:///)?", "", url)}'
```

```

Reply:
`return f'salt://{re.sub(r"^file:(?:/+)?", "", url)}'`

Reply:
> ```python
> return f'salt://{re.sub(r"^file:(?:/+)?", "", url)}'
> ```

If you are going to do zero or more you don't `+` and `?` you just use `*`

```python
return f'salt://{re.sub(r"^file:(?:/*)", "", url)}'
```

Reply:
and if you want to do 0 to 3 `/` then it would be 
```python
return f'salt://{re.sub(r"^file:/{0,3}", "", url)}'
```

or 

```python
return f'salt://{re.sub(r"^file:/{,3}", "", url)}'
```
but I think the explicit `0` is more readable.